android/AndroidSportsPlatform/app/src/main/java/com/example/sportsplatform/viewmodels/SearchViewModelFactory.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.sportsplatform.viewmodels
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+
+class SearchViewModelFactory: ViewModelProvider.NewInstanceFactory() {
+
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+        return SearchViewModel() as T
+    }
```

Comment:
We might create a generic vievmodelfactory class that creates all the factories instead of creating for each.